cloudinit/sources/DataSourceAzure.py

```diff
@@ -1856,6 +1858,7 @@ def load_azure_ds_dir(source_dir):
 @azure_ds_telemetry_reporter
 def generate_network_config_from_instance_network_metadata(
     network_metadata: dict,
+    apply_ip_network_config: bool,
```

Comment:
preferably 
```suggestion
    *, apply_ip_network_config: bool,
```
and pass apply_ip_network_config by keyword arg so it's clear what the parameter is

Reply:
Done!